aaudio/echo/src/main/cpp/echo_audio_engine.cc

```diff
@@ -111,18 +112,23 @@ void EchoAudioEngine::startStreams() {
 }
 
 /**
- * Stops and closes the playback and recording streams
+ * Stops and closes the playback and recording streams.
  */
-void EchoAudioEngine::stopStreams() {
+void EchoAudioEngine::closeAllStreams() {
+
+ /**
+  * Note: The order of events is important here.
+  * The playback stream must be closed before the recording stream. If the recording stream were to
+  * be closed first the playback stream's callback may attempt to read from the recording stream
+  * which would cause the app to crash since the recording stream would be null.
+  */
 
   if (playStream_ != nullptr) {
-    stopStream(playStream_);
```

Comment:
I think it is still good practice for the app to explicitly stop() the stream.
We call stop() in the close() in case they forget.